### DISCUSSION TITLE
Default values as consts?

### DISCUSSION THREAD

**Discussion by sqcsabbey** [simolus3/drift#3157] [Q&A]
I'd like to do something like this:

```
class NtripMountPoints extends TableBase {
  static const defaultSendPosition = false;

  . . . 
  BoolColumn get sendPosition => boolean()
      .withDefault(const Constant(NtripMountPoints.defaultSendPosition))();
  . . .
```

That lets me refer to default values in places like when instantiating Data classes with the variable, instead of hard-coded values, so a change can easily be reflected throughout my code. This falls apart with generated classes, like frozen schemas and schema steps; the consts aren't included or referenced, so the generated code has errors. 

Is there a better way for me to code this?

**Comment by simolus3** [comment]
Classes reconstructed from exported schemas will always be very minimal and only reflect the actual database schema, not the Dart representation associated with it. 

So there's not really a better way to write this that gives you the constant as a static field somewhere. You might be able to use [schema introspection](https://drift.simonbinder.eu/docs/dart-api/schema_inspection/) to extract the `GeneratedColumn.defaultValue` as a `Constant`. That at least gives you a single source of truth, but it's not exactly accessible.

**Reply by sqcsabbey** [reply]
Would it be possible to recognize that it's a const variable, and use its value?

For example, in the json schema dump, generate:

`"default_dart":"const Constant(false)"`

instead of:

`"default_dart":"const Constant(NtripMountPoints.defaultSendPosition)"`

given the code above, namely:

```
class NtripMountPoints extends TableBase {
  static const defaultSendPosition = false;

  . . . 
  BoolColumn get sendPosition => boolean()
      .withDefault(const Constant(NtripMountPoints.defaultSendPosition))();
  . . .
```

**Reply by simolus3** [reply]
Oh, so that's the problem. I understand, but it's also not easy to fix (or at least I don't know how). We see the `const Constant(NtripMountPoints.defaultSendPosition)` as an expression, but despite it being a constant there's no analyzer API (that I'm aware of) that would give us the resolved value of the constant to use.
